src/librustc_trans/mir/analyze.rs

```diff
@@ -109,23 +110,32 @@ impl<'mir, 'a, 'tcx> Visitor<'tcx> for LocalAnalyzer<'mir, 'a, 'tcx> {
                              block: mir::BasicBlock,
                              kind: &mir::TerminatorKind<'tcx>,
                              location: Location) {
-        match *kind {
+        let check = match *kind {
             mir::TerminatorKind::Call {
                 func: mir::Operand::Constant(box mir::Constant {
                     literal: Literal::Value {
-                        value: &ty::Const { val: ConstVal::Function(def_id, _), .. }, ..
+                        value: &ty::Const { val, ty }, ..
                     }, ..
                 }),
                 ref args, ..
-            } if Some(def_id) == self.fx.cx.tcx.lang_items().box_free_fn() => {
+            } => match val {
+                ConstVal::Value(Value::ByVal(PrimVal::Undef)) => match ty.sty {
+                    ty::TyFnDef(did, _) => Some((did, args)),
```

Comment:
Again, only need to check the type here.

Reply:
done